merging 2 rules into 1

parmeet4dev opened the pull request:
Merged two ****byte match set **** rules to one using the HCL2 language by replacing the strings with the Map of lists.
An example is updated on the homepage usage section

Commit:
merging 2 rules into 1

x4121 commented on README.md:
```diff
@@ -20,11 +20,27 @@ Example ACL to add a rate limit to the route `/session` of a given ALB (The ALB
   source                = "github.com/ryte/INF-tf-waf.git//rate_based_rule?ref=v0.3.2"
-  pattern               = "/session"
-  positional_constraint = "STARTS_WITH"
-  metric_name           = "WAFRBRuleMatchSession"
+  allowed_headers = local.allowed_headers
+  metric_name     = "WAFRBRuleMatchSession"
```
Please fix the indentation manually until we have a pre-commit hook implemented for this repo

x4121 commented on rate_based_rule/main.tf:
```diff
@@ -1,19 +1,24 @@
 
 resource "aws_wafregional_rate_based_rule" "url_match_rule" {
-  name        = "wafregional_rate_based_match"
+  name  = "wafregional_rate_based_match"
+  count = length(var.allowed_headers) >= 1 ? 1 : 0
   tags        = local.tags
   metric_name = var.metric_name
```
same here

parmeet4dev replied:
updated the changes.
Thanks for noticing

x4121 reviewed (changes requested):
Otherwise great PR, thanks :+1:

Commit:
fixing indentation

x4121 reviewed (changes requested):
Ah sorry, just remembered: Please update the main Readme with the correct description and changelog

Commit:
updating Readme for rate_based_rule

parmeet4dev commented:
> Ah sorry, just remembered: Please update the main Readme with the correct description and changelog

updated the readme

Commit:
Updated README.md with usage details

parmeet4dev commented:
Updated the main README.md file with the example and changelog

x4121 committed:
Update version tag in Readme

x4121 reviewed (approved):
The version tags in the Readme were not updated, fixed that.

Commit:
updated gitignore, local and output

parmeet4dev commented:
1. updated main.tf to accommodate byte_match_set in rate-based rule based on the count.index
2. updated output.tf to only provide rule id
3. updated locals.tf to add tags

x4121 commented on rate_based_rule/locals.tf:
```diff
@@ -1,3 +1,10 @@
 locals {
-  tags = var.tags
-}
+  tags = {
+    "env"         = "dev"
+    "squad"       = "inf"
+    "product"     = "internal"
+    "type"        = "infra"
+  }
```
Why did you add this? The tags should be passed into the module. Otherwise it will always be `squad = inf, env = dev` no matter who uses it

x4121 commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+.terraform/
+plan*
+*.tfstate
+*.backup
```
Did you add this to the repo by accident?

x4121 committed:
Update branch and fix remaining issues
